src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -170,7 +170,14 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event): vo
                 };
             }
 
-            $arguments[$i] = $payload;
+            if ($argument->metadata->isVariadic()) {
+                if (!(is_array($payload) && array_is_list($payload))) {
+                    throw HttpException::fromStatusCode(422);
+                }
+                array_splice($arguments, $i, count($payload), $payload);
```

Comment:
wouldn't array_merge be more appropriate?

```suggestion
                array_splice($arguments, $i, \count($payload), $payload);
```

Reply:
I wish it work, here I put more details https://github.com/symfony/symfony/pull/54901#discussion_r1695429454